### ISSUE TITLE
renderWithStore seems to be broken

### ISSUE THREAD

**Issue by rfuerst87** [OutWatch/outwatch#310]
Hey Guys

Today I was playing around with the latest master branch (676f94a), trying to upgrade my app from 1.0.0-RC2. This went surprisingly well, except for getting `OutWatch.renderWithStore` to work. I just couldn't get rid of the `NoStoreException`. Either I'm doing something horribly wrong, or this is an actual bug. The following snippet (shamelessly stolen and adapted from #174) demonstrates the issue:

```
object App {

  sealed trait Action
  case object Init extends Action
  case object Add extends Action
  case object Subtract extends Action

  type State = Int
  val initialState = 0

  val reducer = Reducer[Action, State]((state, action) => action match {
    case Add => (state + 1, Observable.empty)
    case Subtract => (state - 1, Observable.empty)
  })

  def main(args: Array[String]): Unit = {
    val root = for {
      store <- Store.get[Action, State]
      root <- IO(
        div(
          button("+", onClick(Add) --> store),
          button("-", onClick(Subtract) --> store),
          p("count = ", child <-- store)
        )
      )
    } yield root

    root
      .flatMap(r => OutWatch.renderWithStore(Init, initialState, reducer, "#content", r))
      .unsafeRunSync()
  }
}
```
Any hints are very appreciated.

**Comment by elyphas** [comment]
Hi, sorry, I can't help much but I think you could take a look a this examples.
https://github.com/OutWatch/outwatch/issues/283
until the people who knows better this, have time to respond

**Comment by cornerman** [comment]
@rfuerst87 Thanks for opening this issue. It definitely highlights, the problems of our singleton store with its `NoStoreException`. The compiler cannot guarantee correctness here.

What happens is that you are mapping over `root`, which will access the `Store` already. This happens before `Outwatch.renderWithStore` is called. But only Outwatch.renderWithStore` creates the `Store`, therefore you get an exception if you try to use it before it is created.

So, what you should do is the following:

```scala
    val root = div(
      Store.get[Action, State].map { store =>
        VDomModifier(
          button("+", onClick(Add) --> store),
          button("-", onClick(Subtract) --> store),
          p("count = ", child <-- store)
        )
      }
    )

    OutWatch.renderWithStore(Init, initialState, reducer, "#content", root).unsafeRunSync()
```

Now, this snippet assures that `Store.get is only called AFTER `OutWatch.renderWithStore`. I give in, that this is not really intuitive. We should make this more explicit. Furthermore, it might help if `renderWithStore` would accept an `IO[VNode]` and not only a `VNode`, this could have helped here.

Hope it helps, let me know if you have further questions :)

**Comment by rfuerst87** [comment]
Many thanks for the quick responses! Indeed passing an `IO` feels more intuitive (at least for me) than working with `VDomModifier`. Especially coming from Version 1.0.0-RC2 where `VNode` was actually of type `IO`.

I'd provide a PR and grab some low hanging fruits by changing the type in `renderWithStore` from `VNode` to `IO[VNode]` if you like. Let me know when I can help :)

**Comment by cornerman** [comment]
I'd provide a PR and grab some low hanging fruits by changing the type in renderWithStore from VNode to IO[VNode] if you like. Let me know when I can help :)

@rfuerst87 That sounds awesome! I think, a PR for this would definitely improve the usability and be less surprising for the user.

**Comment by cornerman** [comment]
Fixed with 1791e9163ae3c075605ae10519dc048bbe030df1, thank you @rfuerst87
